src/H5Dvirtual.c

```diff
@@ -2299,7 +2299,7 @@ H5D__virtual_init(H5F_t *f, const H5D_t *dset, hid_t dapl_id)
 hbool_t
 H5D__virtual_is_space_alloc(const H5O_storage_t H5_ATTR_UNUSED *storage)
 {
-    hbool_t ret_value; /* Return value */
+    hbool_t ret_value = FALSE; /* Return value */
```

Comment:
Seriously?  The compiler has issues with this?   How?

Reply:
Yeah, I was surprised too, but it's not wrong.  :)  It's the evil of macros.  After preprocessing, the function is basically:

```
hbool_t
H5D__virtual_is_space_alloc(const H5O_storage_t __attribute__((unused)) *storage)
{
    hbool_t ret_value;

    if (H5D_init_g || !(H5_libterm_g)) {
        ret_value = 1;
    }
	return (ret_value);
}
```

Reply:
Nope - that ret_value = TRUE is outside of macros and there's no blocks.   The compiler is just wrong. ;-)

Reply:
I was going from the actual preprocessor output.  Here it is unaltered.  Perhaps I should go to bed, but I think clang is right:

```
hbool_t
H5D__virtual_is_space_alloc(const H5O_storage_t __attribute__((unused)) *storage)
{
    hbool_t ret_value;

    { { static hbool_t func_check = 0; if (!func_check) { (__builtin_expect(!((((isdigit((int)(__func__[1])) || isupper((int)(__func__[1]))) && '_' == __func__[2] && '_' == __func__[3] && islower((int)(__func__[4]))) || ((isdigit((int)(__func__[2])) || isupper((int)(__func__[2]))) && '_' == __func__[3] && '_' == __func__[4] && islower((int)(__func__[5]))) || ((isdigit((int)(__func__[3])) || isupper((int)(__func__[3]))) && '_' == __func__[4] && '_' == __func__[5] && islower((int)(__func__[6])))) && "Function naming conventions are incorrect - check H5_IS_API|PUB|PRIV|PKG macros in " "H5private.h (this is usually due to an incorrect number of underscores)"), 0) ? __assert_rtn(__func__, "/Users/sean/external/hdf5/src/H5Dvirtual.c", 2304, "(((isdigit((int)(__func__[1])) || isupper((int)(__func__[1]))) && '_' == __func__[2] && '_' == __func__[3] && islower((int)(__func__[4]))) || ((isdigit((int)(__func__[2])) || isupper((int)(__func__[2]))) && '_' == __func__[3] && '_' == __func__[4] && islower((int)(__func__[5]))) || ((isdigit((int)(__func__[3])) || isupper((int)(__func__[3]))) && '_' == __func__[4] && '_' == __func__[5] && islower((int)(__func__[6])))) && \"Function naming conventions are incorrect - check H5_IS_API|PUB|PRIV|PKG macros in \" \"H5private.h (this is usually due to an incorrect number of underscores)\"") : (void)0); func_check = 1; } };; if (H5D_init_g || !(H5_libterm_g)) {






    ret_value = 1;

    ; } return (ret_value); }
}
```

Reply:
Ah!   That's really funny, in a way - it looks like a bug in the FUNC_ENTER_PACKAGE_NOERR macro!   That macro (in src/H5private.h) is supposed to make certain that the package has been initialized before executing the code in the routine.  It's currently doing that by wrapping the 'if ( <package init> ... ) ' around the function's code, but it really should be an assert( <package init>) in the macro, instead of an 'if' statement.   I think it's harmless to initialize the ret_value here, so I don't want to block your PR, but I'll also create a PR that fixes the macro. 👍

Reply:
Nice find!  This is the argument against blindly initializing things: by leaving things uninitialized, the compiler can often find codepaths where something is used uninitialized.

It could be that I have "suppressed" other interesting finds too. :)